### PR TITLE
Bookstore API PR

### PR DESCRIPTION
**Pull request by kkasaija**
- [x] Used useSelector to consume the state.
- [x] Imported Redux Provider and store in the main component of React app.
- [x] Imported Redux useDispatch.
- [x] Used `fetch` with `createAsyncThunk` for making API calls.
- [x] Fetched book list from Bookstore API on Books page load
- [x] Refactored `add book` and `remove book` features to persist changes in the server.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,47 +1,58 @@
...
-});
+// Action Creators
+const addBook = (book) => async (dispatch) => {
+  await fetch(Url, {
+    method: 'POST',
+    body: JSON.stringify(book),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
```
Comment (Munsa1) [inline src/redux/books/books.js]: - [ ] It is a requirement that `createAsyncThunk` is used in the project for creating thunk action creators: Kindly fix this issue
Example


`
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
`

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Kasaija-Kenneth , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._
- [ ] `createAsyncThunk` is to be used in the project for creating thunk action creators: Take a look at the screenshot below
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Kasaija-Kenneth , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

_**Kindly check an optional comment I left under the review**_ :pray: 

## Highlights

-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 
-   App worked as expected. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkasaija** [comment]
Thank you

**Pull request merged**